File: 948/main.cpp
#include<iostream>
#include<string>
using namespace std;

int fib(int num) {
  if (num <= 1) {
    return num;
  }
  else {
    return fib(num - 1) + fib(num - 2);
  }
}

int main() {
  int testcases;
  cin >> testcases;
  for (int i = 0; i < testcases; i++) {
    int num;
    string binary = "";
    cin >> num;
    int fib_num = fib(num);
    while (fib_num > 0) {
      int digit = fib_num % 2;
      binary = to_string(digit) + binary;
      fib_num = fib_num / 2;
    }
    cout << num << " = " << binary << " (fib)" << endl;
  }
}

File: 948/main.py
def main():
    fib = [1, 2]
    while fib[-1] < 100000000:
        fib.append(fib[-1] + fib[-2])

    for _ in range(int(input())):
        num = int(input())
        print(f"{num} = ", end="")

        found = 0
        for f in fib[::-1]:
            if num >= f:
                num -= f
                found = 1
                print(1, end="")

            elif found:
                print(0, end="")
        print(" (fib)")


if __name__ == "__main__":
    main()
